lib/wallet/account/hot-wallet-private-key.ts

```diff
@@ -116,29 +120,50 @@ export class HotWalletPrivateKey implements IWallet {
       throw new Error("No transaction to sign");
     }
 
-    // Replace the input script with the provided script
-    if (options?.scriptHex) {
-      const replaceIndex = pending.transaction.inputs.findIndex(
-        (input) => input.signatureScript === "",
-      );
-      if (replaceIndex === -1) {
-        throw new Error("No input to replace");
+    const signed = await this.signTx(pending.transaction, options?.scripts);
+    return (await this.rpcClient.submitTransaction({ transaction: signed }))
+      .transactionId;
+  }
+
+  // NOTE: This method does not support signing with multiple keys
+  async signTx(tx: Transaction, scripts?: ScriptOption[]) {
+    if (scripts) {
+      for (const script of scripts) {
+        await this.signTxInputWithScript(tx, script);
       }
+      return await signTransaction(tx, [this.getPrivateKey()], false);
```

Comment:
This line can be remove the routine will be identical.

Reply:
https://github.com/forbole/kastle/pull/22/commits/8b988b509dea2961248604d38d00cd36da2d12e5